src/utilities/frame/hooks.tsx

```diff
@@ -0,0 +1,27 @@
+import {useContext} from 'react';
+import {useAppBridge} from '../app-bridge';
+import {FrameContext} from './context';
+
+export function useFrame() {
+  const frame = useContext(FrameContext);
+  const appBridge = useAppBridge();
+
+  if (!frame && !appBridge) {
+    throw new Error(
+      'Your application must be wrapped in an <Frame> component to use: Loading, Toast, or ContextualSaveBar. See https://polaris.shopify.com/components/structure/frame for implementation instructions.',
+    );
+  }
+
+  return (
+    frame || {
```

Comment:
If we're specifying a default here, then we might as well use that default for FrameContext, and thus we don't need to check if value is undefined and potentially throw an error (as it will always be set).